model_training/data_preparation/measurement_utils.py

```diff
@@ -12,25 +12,63 @@
 }
 
 
-def preprocess_measurement_data(path: Path) -> pd.DataFrame:
+def preprocess_measurement_data(
+    path: Path, min_per_layer_measurement: int
+) -> pd.DataFrame | None:
     """Read and preprocess measurement data from CSV file.
 
     For each layer, we have N records of power and latency.
     We get average for power and latency for each layer over N cycles.
 
     Args:
         path: Path to csv file
+        min_per_layer_measurement: Minimum number of samples
+            of power and runtime values to be present for each layer
 
     Returns:
         Parse csv data as pandas dataframe.
     """
     df = pd.read_csv(path)
-    df = process_measurement_data(df)
-    df.rename(columns=RENAME_COLUMNS, inplace=True)
-    return df.loc[:, SELECT_COLUMNS]
+    if verify_per_layer_measurements(df, min_per_layer_measurement):
```

Comment:
We can swap the operations around. If we calculate the statistics for each layer first (mean, std, count), we can then simply check count for which layers is below the threshold.

Reply:
Yes, this is addressed.